### DISCUSSION TITLE
How to create Metadata with union type?

### DISCUSSION THREAD

**Discussion by SymphonyNineth** [nartc/mapper#475] [Q&A] [answered]
I'm using automapper with Pojo strategy. I have to define metadata for it, but "postalCode" field have a union type.

 ```
     PojosMetadataMap.create<FrontendData>("FrontendData", {
        address: String,
        postalCode: null | number,
        latitude: Number,
        longitude: Number,
     });
```

Unfortunately the syntax null postalCode: null | number does not work. I have looking in documentation but there was nothing about using union types with this strategy - https://automapperts.netlify.app/docs/strategies/pojos/. Is there a way to do that?

**Comment by nartc** [comment, marked as answer]
As of the moment, there's no union support for metadata. `number | null` is still Number in this case. If the source `postalCode` is `null` then it will be mapped to `null`, if the source `postalCode` is `undefined` then it will be mapped to `undefined`. You can use `undefinedSubstitution(null)` to convert it to `null` if the source is `undefined`

**Reply by SymphonyNineth** [reply]
Hi. Thank you for your time.
So i can just specify type `number` in metadata, even if in the interface the type is `null | number` and it will work.
And can i specify `undefinedSubstitution` for every field?

**Reply by nartc** [reply]
There's no global member map configuration :( `afterMap()` can be used to say like: after I map, I want to turn all `undefined` to `null` by a simple loop

**Reply by SymphonyNineth** [reply]
That will work thank you very much.
And one last question - When some field is present in destination object and absent in source object i get an error. Can i tell it to just ignore the fields that are absent in source?

**Reply by SymphonyNineth** [reply]
And actually last question - What can this error mean  `Mapping is not found for Symbol(Label) and function String() { [native code] }` ??
Sorry if I'm taking too much of your time.
